packages/next/src/build/webpack/plugins/define-env-plugin.ts

```diff
@@ -159,7 +159,10 @@ export function getDefineEnv({
     'process.turbopack': isTurbopack,
     'process.env.TURBOPACK': isTurbopack,
     // TODO: enforce `NODE_ENV` on `process.env`, and add a test:
-    'process.env.NODE_ENV': dev ? 'development' : 'production',
+    'process.env.NODE_ENV':
+      dev || config.experimental.allowDevelopmentBuild
```

Comment:
Since this is forcing the development build should we call it that to reflect it e.g. `forceReactDevelopmentMode`?

Reply:
I initially went with `force`, but since I'm requiring you to also specify `NODE_ENV=development`, I figured "allow" made more sense.